lib/cylc/network/port_scan.py

```diff
@@ -232,6 +233,28 @@ def scan_many(items, timeout=None, updater=None):
     return results
 
 
+def re_compile_filters(patterns_owner=None, patterns_name=None):
+    """Compile regexp for suite owner and suite name scan filters.
+
+    Arguments:
+        patterns_owner (list): List of suite owner patterns
+        patterns_name (list): List of suite name patterns
+
+    Returns (tuple):
+        A 2-element tuple in the form (cre_owner, cre_name). Either or both
+        element can be None to allow for the default scan behaviour.
+    """
+    cres = {'owner': None, 'name': None}
+    for label, items in [('owner', patterns_owner), ('name', patterns_name)]:
+        if items:
+            cres[label] = r'\A(?:' + r')|(?:'.join(items) + r')\Z'
```

Comment:
I think this is probably the right way to go but It's worth noting this is a change in behaviour, partial patterns used to match.

Reply:
Added an entry to CHANGES.md and modified the `cylc scan` syntax on `cylc run` daemonize.